### PR TITLE
add Restaurant API

### PR DESCRIPTION
**Pull request by luxd**
1. build.gradle add 1 dependency
2. add RestaurantController
3. add Restaurant model, but it is not used actually
4. add RestaurantTest, to run the test need to modify the test package name
5. need add properties record, not push it yet

### CODE REVIEW

**Comment by j-bro** [comment]
The point of putting the API keys in the ignored file `services.properties` is that they won't be publicly shared on GitHub. You should get the file from Slack and keep it locally but it will stay ignored. We will need to create new keys since these ones are now shared with the world because of this commit. You should use the `ServicesConfig` interface which you can also `@Autowire` in a controller, it will read properties from the `services.properties` file and expose property getter methods such as `googleMapsApiKey()`. Just add any extra keys you may need to that file instead of `application.properties`.

File: backend/src/main/java/com/soen487/project/backend/controllers/RestaurantController.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+/**
+ * Created by ericl on 4/1/2017.
+ */
+@RestController
+public class RestaurantController {
+
+
+    @Autowired
+    private Environment env;
```
Comment (j-bro) [inline backend/src/main/java/com/soen487/project/backend/controllers/RestaurantController.java]: `private ServicesConfig servicesConfig;`

File: backend/src/main/java/com/soen487/project/backend/controllers/RestaurantController.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    @RequestMapping(value = "/restaurant", method = GET)
+    @ResponseBody
+    public String getBestRestaurant(@RequestParam Double longitude, @RequestParam Double latitude, @RequestParam int topN) {
+
+        Map<String, String> vars = new HashMap<>();
+        vars.put("location", Double.toString(latitude)+","+Double.toString(longitude)); //lat + long
+        vars.put("radius", "5000");
+        vars.put("types","restaurant");
+        vars.put("sensor","true");
+        vars.put("key",env.getProperty("googlemaps.api.key")); //get google developer api key
```
Comment (j-bro) [inline backend/src/main/java/com/soen487/project/backend/controllers/RestaurantController.java]: `vars.put("key", servicesConfig.googleMapsApiKey());`

**Comment by luxd** [comment]
Updated the RestaurantController.
But actually after change I can't get it passed in the unit test. The google api key is always null. I don't know why.

**Comment by j-bro** [comment]
@luxd did you add the `services.properties` file from Slack to `src/main/resources/services.properties`?

**Comment by luxd** [comment]
@j-bro Yes, I did. If I use hard code for the key value or use Env to get the value from properties file, both work.
If the updated one can work on your computer, that is fine. We can make the project running first, later on we can find the reason for this problem.

**Comment by j-bro** [comment]
I just tried it locally and it works fine for me. One last thing you need to add is the JSON dependency to build.gradle as so: `compile 'com.vaadin.external.google:android-json:0.0.20131108.vaadin1'`. With that added we can merge.

**Comment by j-bro** [comment]
I'll just merge this and add the changes in master myself.

**Pull request merged**